File: Client/Objects/Units/Units/IUnit.cs
using System.Collections.Generic;
using Godot;

public interface IUnit
{

    public Vector3 Position { get; set; }
    public Vector2I PositionI { get; set; }
    public UnitData Data { get; set; }
    public Node3D TileNode { get; set; }
    
    public void InitUnit();
    
    public void Skill(Vector2I position, string skillName);
    public Dictionary<string, ISkill> Skills { get; set; }
    
    public void Move(TerrainInfo fromPos, TerrainInfo toPos, bool message = false);
    public IMovement Movement { get; set; }

    void NewTurn();

    void AddToSkillBuffer(int skillId, ICombatAction action);
    void RemoveFromSkillBuffer(int skillId);
    
    void Inflict(int skillId, ICombatAction action, int Turns);
    void RemoveInflict(int skillId);
    
    void Damage(float amount);
    void Heal(float amount);

}

public class InflictAction
{
    public ICombatAction Action;
    public int Turns;
}

File: server/internal/server/objects/units/units/IUnit.go
package units

import (
	"server/internal/server"
	"server/internal/server/combatDSL"
	"server/internal/server/objects"
	"server/internal/server/objects/units/Movement"
	"server/internal/server/objects/units/Skills"
	"server/internal/server/objects/units/units/util"
	"server/pkg/packets"
)

type IUnit interface {
	NewUnit(*util.UnitData, *server.Client, *server.Client, Skills.GameSkillService)
	Position() objects.Vector3
	SetPosition(*objects.Vector3)
	Data() *util.UnitData

	Skills() []Skills.ISkill
	Attack(IUnit)

	NewTurn()

	Movement() *Movement.IMovement
	Move(path []*packets.HexPositionMessage) bool

	Damage(Amount int64)
	Heal(Amount int64)

	AddToSkillBuffer(skillID int32, action combatDSL.CombatAction)
	RemoveFromSkillBuffer(skillId int32)
	Inflict(skillID int32, action combatDSL.CombatAction, turns int)
	RemoveInflict(skillID int32)
}

type InflictAction struct {
	Action combatDSL.CombatAction
	Turns  int
}
